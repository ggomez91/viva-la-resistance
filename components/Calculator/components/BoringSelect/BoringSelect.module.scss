.BoringSelect {
  text-align: left;
  margin: 10px 0;
}
