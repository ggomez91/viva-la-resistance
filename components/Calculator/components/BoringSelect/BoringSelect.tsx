import React from "react";
import styles from "./BoringSelect.module.scss";
import _ from "lodash";
import colors from "../../domain/colors";

type BoringSelectProps = {
  color: string;
  setColor: (color: string) => void;
  title: string;
};

/**
 * Renders a bootstrap select with the known colors.
 */
const BoringSelect: React.FC<BoringSelectProps> = ({
  color,
  setColor,
  title,
}) => {
  return (
    <div className={styles.BoringSelect}>
      <div className="form-group">
        <label htmlFor={title}>{_.capitalize(title)}</label>
        <select
          value={color}
          className="form-control"
          id={title}
          onChange={(e) => {
            setColor(e.target.value);
          }}
        >
          <option value={null}>none</option>
          {colors.map((color) => (
            <option key={color} value={color}>
              {color}
            </option>
          ))}
        </select>
      </div>
    </div>
  );
};

export default BoringSelect;
