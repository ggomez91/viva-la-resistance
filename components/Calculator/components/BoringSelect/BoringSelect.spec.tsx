import React from "react";
import renderer from "react-test-renderer";
import BoringSelect from "./BoringSelect";

describe("BoringSelect", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <BoringSelect color={"red"} setColor={jest.fn()} title="field name" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
