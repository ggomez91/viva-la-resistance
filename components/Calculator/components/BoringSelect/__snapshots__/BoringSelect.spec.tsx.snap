// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoringSelect renders correctly 1`] = `
<div>
  <div
    className="form-group"
  >
    <label
      htmlFor="field name"
    >
      Field name
    </label>
    <select
      className="form-control"
      id="field name"
      onChange={[Function]}
      value="red"
    >
      <option
        value={null}
      >
        none
      </option>
      <option
        value="pink"
      >
        pink
      </option>
      <option
        value="silver"
      >
        silver
      </option>
      <option
        value="gold"
      >
        gold
      </option>
      <option
        value="black"
      >
        black
      </option>
      <option
        value="brown"
      >
        brown
      </option>
      <option
        value="red"
      >
        red
      </option>
      <option
        value="orange"
      >
        orange
      </option>
      <option
        value="yellow"
      >
        yellow
      </option>
      <option
        value="green"
      >
        green
      </option>
      <option
        value="blue"
      >
        blue
      </option>
      <option
        value="violet"
      >
        violet
      </option>
      <option
        value="gray"
      >
        gray
      </option>
      <option
        value="white"
      >
        white
      </option>
    </select>
  </div>
</div>
`;
